src/Middleware/Authentication/GitHubAppTokenProvider.cs

```diff
@@ -0,0 +1,79 @@
+using System.Net.Http.Headers;
+using System.Security.Cryptography;
+using System.Text.Json;
+using GitHub.Octokit.Client.Middleware.Options;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Tokens;
+
+namespace GitHub.Octokit.Client.Authentication;
+
+/// <summary>
+/// This class is responsible for creating, signing and refreshing access tokens
+/// </summary>
+public class GitHubAppTokenProvider : IGitHubAppTokenProvider
```

Comment:
I picked the `GitHubAppTokenProvider` and `IGitHubAppTokenProvider` naming here, as it used to be `AccessTokenProvider`, which I felt clashed with Kiota's `IAccessTokenProvider` naming/interface, which this does _not_ implement. Please let me know if you have thoughts or better ideas on naming here!

Reply:
This just doesn't feel right... perhaps because it's not consistent across all provider naming schemes or it feels redundant with the namespace.  

My preference would be to keep it as it was, and just name the interface `IGitHubAppTokenProvider` or rename them all for consistency (including the test files).  Perhaps in the future we'll implement other auth providers? IDK. 

I understand the reasoning behind the change, I think my only critique is go all or none with the rename IMO.

Reply:
Also, I am wondering if we should move Authentication back to a root level folder (like Go), again, for consistency.

Reply:
Agreed about it not quite feeling right. My main concern with the way it was, was I found it very confusing that our naming scheme matched Kiota's naming scheme for an interface that we didn't implement. Perhaps we can pair on this tomorrow? 

I don't have a huge preference on moving Auth code back to the root out of middleware...I'm happy to go either way, and make a decision to potentially revise later.